hexrd/ui/calibration/wppf_options_dialog.py

```diff
@@ -420,6 +421,136 @@ def create_wppf_params_object(self):
+                rid = list(_rqpDict[mat.unitcell.latticeType][0])
+
+                name = _lpname[rid]
+                for i, (n, l) in enumerate(zip(name, lp)):
+                    nn = p+'_'+n
+
+                    """
+                    first 3 are lengths, next three are angles
+                    """
+                    if(rid[i] <= 2):
+                        params.add(nn, value=l, lb=l-0.05,
+                                   ub=l+0.05, vary=False)
+                    else:
+                        params.add(nn, value=l, lb=l-1.,
+                                   ub=l+1., vary=False)
+
+                """
+                now adding the atom positions and 
+                occupancy
+                """
+                atom_pos = mat.unitcell.atom_pos[:, 0:3]
+                occ = mat.unitcell.atom_pos[:, 3]
+
+                atom_type = mat.unitcell.atom_type
+                atom_label = _getnumber(atom_type)
+                
+                #self.atom_label = atom_label
+
+                """
+                now for each atom type append the fractional
+                coordinates, occupation fraction and debye-waller
+                factors to the list of parameters
+                """
+                for i in range(atom_type.shape[0]):
+                    Z = atom_type[i]
+                    elem = constants.ptableinverse[Z]
+                    # x-coordinate
+                    nn = p+'_'+elem+str(atom_label[i])+'_x'
+                    params.add(
+                        nn, value=atom_pos[i, 0],
+                        lb=0.0, ub=1.0,
+                        vary=False)
+
+                    # y-coordinate
+                    nn = p+'_'+elem+str(atom_label[i])+'_y'
+                    params.add(
+                        nn, value=atom_pos[i, 1],
+                        lb=0.0, ub=1.0,
+                        vary=False)
+
+                    # z-coordinate
+                    nn = p+'_'+elem+str(atom_label[i])+'_z'
+                    params.add(
+                        nn, value=atom_pos[i, 2],
+                        lb=0.0, ub=1.0,
+                        vary=False)
+
+                    # occupation
+                    nn = p+'_'+elem+str(atom_label[i])+'_occ'
+                    params.add(nn, value=occ[i],
+                               lb=0.0, ub=1.0,
+                               vary=False)
+
+
+                    if(mat.unitcell.aniU):
+                        U = mat.unitcell.U
+                        for j in range(6):
+                            nn = p+'_'+elem + \
+                                str(atom_label[i])+'_'+_nameU[j]
+                            params.add(
+                                nn, value=U[i, j],
+                                lb=-1e-3,
+                                ub=np.inf,
+                                vary=False)
+                    else:
+                        nn = p+'_'+elem+str(atom_label[i])+'_dw'
```

Comment:
Note that in python >= 3.6, you can use f-strings like this:
```python
nn = f'{p}_{elem}{atom_label[i]}_dw'
```

and it might be a little simpler.

Reply:
ok got it.

Reply:
yes, endorse!